www/guides/react.md

```diff
@@ -0,0 +1,350 @@
+
+Start by enabling [Hot Module Replacement](https://www.snowpack.dev/#hot-module-replacement), a requirement for Fast Refresh. Snowpack automatically refreshes to show changes in your code. Hot Module Replacement only updates the components that changed without refreshing the whole page. You can enable it for React with a small snippet of code in `index.jsx`.
+
+Head to `index.jsx` and add this snippet at the bottom:
+
+```diff
+  ReactDOM.render(
+    <React.StrictMode>
+      <App />
+    </React.StrictMode>,
+    document.getElementById('root'),
+  );
++ // Hot Module Replacement (HMR) - Remove this snippet to remove HMR.
++ // Learn more: https://www.snowpack.dev/#hot-module-replacement
++ if (import.meta.hot) {
++   import.meta.hot.accept();
++ }
+```
+
+Now when you change `App.jsx` the changes show without the whole page refreshing.
+
+<div class="frame"><img src="/img/guides/react/hmr.gif" alt="GIF showing code side by side with the app. A change in made to App.jsx and it shows immediately when the file is changed. The counter keeps counting uninterrupted." class="screenshot"/></div>
+
+The counter on the page still resets to 0, even though the page itself hasn't fully refreshed. This can be annoying if you're trying to debug state. React Fast Refresh solves this.
+
+Installing a plugin involves installing the module and then enabling it in the Snowpack configuration file.
+
+Install the module first by running.
+
+```bash
+npm install @snowpack/plugin-react-refresh --save-dev
+```
+
+To enable it head to `snowpack-config.js` and add it to the array of plugins
+
+```diff
+  module.exports = {
+    mount: {
+      public: '/',
+      src: '/_dist_',
+    },
+-   plugins: []
++   plugins: ['@snowpack/plugin-react-refresh'],
+  };
+```
+
+> 💡 Tip: you'll need to restart Snowpack for the configuration changes to take effect
+
+Now make an edit to `App.jsx`. If this worked you'll see it without the counter stopping or losing its value.
+
+<div class="frame"><img src="/img/guides/react/react-fast-refresh.gif" alt="GIF showing code side by side with the app. A change in made to App.jsx and it shows immediately when the file is changed. The counter keeps counting uninterrupted." class="screenshot"/></div>
+
+## Going further
+
+Great job! You're now ready to build the React project of your dreams with Snowpack.
```

Comment:
Good to call out that this is the end. "Congratulations! You've reached the end of our guide, and are now ready to..."